libqtile/backend/wayland/window.py

```diff
@@ -409,7 +409,12 @@ def floating(self) -> bool:
     @floating.setter
     def floating(self, do_float: bool) -> None:
         if do_float and self._float_state == FloatStates.NOT_FLOATING:
-            if self.group and self.group.screen:
+            if (
+                self.group
+                and self.group.screen
+                and self.float_x is not None
+                and self.float_y is not None
+            ):
```

Comment:
Can you make this check into a function that returns a boolean since we use it twice? That will also force us to give it a reasonable name.

Reply:
Sure, I just checked around a little and found that `self.group and self.group.screen` is repeated quite often so might also be a candidate for a function.

But for this function you are suggesting (I would call it `has_float_offsets`?): should it include only `self.float_x is not None and self.float_y is not None` or also `self.group and self.group.screen`?

The only place where these float offsets are set is in implementations of `_Window.place`, so `is_placed` would be another possible name for the function, though there is one implementation of `place` (`LayerStatic`) where this is not true.

Should this function be added to `_Window` or `Window`?

Reply:
Seems like Window, since that's the only place it's used. 

I think starting with one that includes all four of the tests in your if is good. We can refactor beyond that in future patches.

Reply:
I have updated it, but mypy is complaining now because it doesn't see that we have made sure that screen can't be `None`.